tests/test_taskFactory.py

```diff
@@ -0,0 +1,162 @@
+
+    def testAllOptions(self):
+        butler = self._tempButler()
+        self.factory.makeTask(taskClass=self.tracker, label="no-name", config=self._alteredConfig(),
+                              overrides=self._overrides(), butler=butler)
+        catalog = butler.get("fakeInitInput")  # Copies of _dummyCatalog are identical but not equal
+        # Overrides ignored
+        self.tracker.assert_called_with(config=self._alteredConfig(),
+                                        initInputs={'initInput': catalog},
+                                        name="no-name")
+
+    # Can't test all 14 remaining combinations, but the 6 pairs should be enough coverage
+
+    def testNameConfig(self):
+        self.factory.makeTask(taskClass=self.tracker, label="no-name", config=self._alteredConfig(),
+                              overrides=None, butler=None)
+        self.tracker.assert_called_with(config=self._alteredConfig(), initInputs=None, name="no-name")
+
+    def testNameOverrides(self):
+        self.factory.makeTask(taskClass=self.tracker, label="no-name", config=None,
+                              overrides=self._overrides(), butler=None)
+        config = self._alteredConfig()
+        self._overrides().applyTo(config)
+        self.tracker.assert_called_with(config=config, initInputs=None, name="no-name")
+
+    def testNameButler(self):
+        butler = self._tempButler()
+        self.factory.makeTask(taskClass=self.tracker, label="no-name", config=None,
+                              overrides=None, butler=butler)
+        catalog = butler.get("fakeInitInput")  # Copies of _dummyCatalog are identical but not equal
+        self.tracker.assert_called_with(config=FakeConfig(),
+                                        initInputs={'initInput': catalog},
+                                        name="no-name")
+
+    def testConfigOverrides(self):
+        self.factory.makeTask(taskClass=self.tracker, label=None, config=self._alteredConfig(),
+                              overrides=self._overrides(), butler=None)
+        # Overrides ignored
+        self.tracker.assert_called_with(config=self._alteredConfig(), initInputs=None, name=None)
+
+    def testConfigButler(self):
+        butler = self._tempButler()
+        self.factory.makeTask(taskClass=self.tracker, label=None, config=self._alteredConfig(),
+                              overrides=None, butler=butler)
+        catalog = butler.get("fakeInitInput")  # Copies of _dummyCatalog are identical but not equal
+        self.tracker.assert_called_with(config=self._alteredConfig(),
+                                        initInputs={'initInput': catalog},
+                                        name=None)
+
+    def testOverridesButler(self):
+        butler = self._tempButler()
+        self.factory.makeTask(taskClass=self.tracker, label=None, config=None,
+                              overrides=self._overrides(), butler=butler)
+        config = self._alteredConfig()
```

Comment:
I think this should be `config = FakeConfig()`, since we don't pass the alteredConfig at all. I think it doesn't matter in practice here, but if someone were to add another config option, it might?

And the same in `testNameOverrides`, now that I notice it.